README.md

```diff
@@ -87,27 +113,34 @@ NOTE: Any key can be prefixed with a ```!``` to overwrite the config value, rath
     "test_steps": [
         "some command or action to run"
     ],
-    // These will be built before my-project, and transitive dependencies will be followed.
+
+    // These will be built before my-project, and transitive dependencies will be followed. Alias: upstream
     "dependencies": [
         {
             "name": "my-lib",
             "revision": "branch-or-commit-sha"
         }
     ],
-    // These will be built when a downstream build is requested
+    // These will be built when a downstream build is requested. Alias: downstream
     "consumers": [
         {
             "name": "my-downstream-project"
         }
     ],
     // These are special, much like CMake's IMPORTED targets, the builder must know about them or they must be
     // defined by a script. For examples, look in builder/imports. Just like dependencies, transitive imports
-    // will be resolved.
+    // will be resolved. If no special configuration is required, these can just be submodules or other repos
     "imports": [
         "s2n"
     ],
-    // Configuration differences per host
+
+    // Per-environment overrides
+    // Overrides are applied per host, per target/architecture, and per compiler/version. Any top-level config 
+    // value can be overridden from within these override sections, see below for examples.
+
+    // Configuration differences per host (the machine/image the build runs on)
```

Comment:
something about whether or not you need to include each host you want to test on or only including the ones you don't want with "enabled": false